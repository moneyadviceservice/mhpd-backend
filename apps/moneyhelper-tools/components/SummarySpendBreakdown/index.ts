export type { Props, PieData } from './SummarySpendBreakdown';
export { SummarySpendBreakdown } from './SummarySpendBreakdown';
