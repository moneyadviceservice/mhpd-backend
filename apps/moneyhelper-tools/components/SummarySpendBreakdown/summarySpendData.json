[
  {
    "name": "Household bills",
    "value": 875,
    "percentage": 40,
    "colour": "#007a91",
    "url": "/cost-tab"
  },
  {
    "name": "Living costs",
    "value": 771.28,
    "percentage": 35,
    "colour": "#f56727",
    "url": "/cost-tab"
  },
  {
    "name": "Finance & Insurance",
    "value": 100,
    "percentage": 10,
    "colour": "#bc0061",
    "url": "/cost-tab"
  },
  {
    "name": "Family & Friends",
    "value": 100,
    "percentage": 10,
    "colour": "#879a24",
    "url": "/cost-tab"
  },
  {
    "name": "Travel",
    "value": 10,
    "percentage": 2.5,
    "colour": "#000d3d",
    "url": "/cost-tab"
  },
  {
    "name": "Leisure",
    "value": 10,
    "percentage": 2.5,
    "colour": "#0b4a6d",
    "url": "/cost-tab"
  }
]
