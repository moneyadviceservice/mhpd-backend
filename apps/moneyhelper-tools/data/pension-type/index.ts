export { pageTitles } from './page-titles';
export { landingText } from './landing';
