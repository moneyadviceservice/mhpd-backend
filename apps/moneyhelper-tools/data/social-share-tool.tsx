import { Icon, IconType } from '@maps-digital/shared/ui';

export const shareToolContent = {
  title: {
    en: 'Share this tool',
    cy: 'Rhannwch yr offeryn hwn',
  },
  items: [
    {
      name: 'email',
      svg: <Icon type={IconType.MAIL} fill={'fill-current'} />,
    },
    {
      name: 'facebook',
      svg: <Icon type={IconType.FACEBOOK} fill={'#3B5998'} />,
    },
    {
      name: 'twitter',
      svg: <Icon type={IconType.TWITTER} fill={'#1DA1F2'} />,
    },
  ],
};

export const ENCODED_BODY =
  'Copy%20and%20paste%20the%20below%20URL%20in%20your%20browser.';

export const ENCODED_NEW_LINE = '%0D%0A';
