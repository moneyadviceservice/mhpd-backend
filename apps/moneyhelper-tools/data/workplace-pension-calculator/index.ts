export { breadcrumbs } from './breadcrumbs';
export { stepData, pageData } from './pension-data';
export { validatePensionInputs, validateAge } from './pension-validation';
export type { Fields, Step } from './pension-data';
export type { ErrorField } from './pension-validation';
