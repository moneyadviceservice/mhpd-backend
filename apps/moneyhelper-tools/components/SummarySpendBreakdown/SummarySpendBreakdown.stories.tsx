import { StoryFn } from '@storybook/react';
import { SummarySpendBreakdown, Props } from './SummarySpendBreakdown';
import data from './summarySpendData.json';

const StoryProps = {
  title: 'Components/TOOLS/SummarySpendBreakdown',
  component: SummarySpendBreakdown,
};

const Template: StoryFn<Props> = (args) => <SummarySpendBreakdown {...args} />;

export const Default = Template.bind({});
Default.args = {
  data: data,
};

const budgetBuilderData = [
  {
    name: 'Family & Friends',
    value: 100,
    hasTick: true,
    percentage: 10,
    colour: '#879a24',
    url: '/cost-tab',
  },
  {
    name: 'Travel',
    value: 100,
    isEstimate: true,
    percentage: 2.5,
    colour: '#000d3d',
    url: '/cost-tab',
  },
];

export const BudgetBuilder = Template.bind({});
BudgetBuilder.args = {
  data: budgetBuilderData,
};

export default StoryProps;
