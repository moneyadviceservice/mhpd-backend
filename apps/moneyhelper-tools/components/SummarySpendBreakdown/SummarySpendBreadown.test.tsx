import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { SummarySpendBreakdown } from './SummarySpendBreakdown';
import data from './summarySpendData.json';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

describe('SummarySpendBreakdown component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <SummarySpendBreakdown data={data} heading="Spending Breakdown" />,
    );
    expect(container).toMatchSnapshot();
  });
  it('renders correctly with tick', () => {
    const { container } = render(
      <SummarySpendBreakdown
        data={[
          {
            name: 'Family & Friends',
            value: 100,
            hasTick: true,
            percentage: 10,
            colour: '#879a24',
            url: '/cost-tab',
          },
        ]}
        heading="Spending Breakdown"
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it('renders correctly as estimate', () => {
    const { container } = render(
      <SummarySpendBreakdown
        data={[
          {
            name: 'Travel',
            value: 100,
            isEstimate: true,
            percentage: 2.5,
            colour: '#000d3d',
            url: '/cost-tab',
          },
        ]}
        heading="Spending Breakdown"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
