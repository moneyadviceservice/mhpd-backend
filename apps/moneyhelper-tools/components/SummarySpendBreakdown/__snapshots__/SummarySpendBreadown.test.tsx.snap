// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummarySpendBreakdown component renders correctly 1`] = `
<div>
  <div
    class="pt-4 lg:pl-7 lg:pt-0 lg:pb-0"
  >
    <h5
      class="text-xl md:text-2xl text-gray-800 font-bold px-4 lg:p-0"
    >
      Spending Breakdown
    </h5>
    <div
      class="flex items-center lg:flex-row flex-col"
    >
      <div
        class="lg:w-4/12 print:hidden w-3/4 px-4 lg:p-0 flex justify-center"
      >
        <div
          class="h-0 pt-[100%] overflow-hidden relative flex-none my-10 w-full"
        >
          <div
            class="after:absolute after:bg-white after:rounded-full after:top-2/4 after:left-2/4 after:-translate-y-2/4 after:-translate-x-2/4 after:h-[55%] after:w-[55%] rounded-full w-full h-full absolute left-0 right-0 top-0 bottom-0"
          />
        </div>
      </div>
      <div
        class="w-full lg:w-8/12"
      >
        <table
          class="w-full lg:text-lg border-separate border-spacing-2 lg:border-spacing-5 lg:ml-10"
        >
          <thead>
            <tr>
              <th
                class="lg:w-9/12"
              >
                <span
                  class="sr-only"
                >
                  Category
                </span>
              </th>
              <th
                class="hidden lg:w-1/12"
              >
                <span
                  class="sr-only"
                >
                  Percentage
                </span>
              </th>
              <th
                class="hidden lg:w-1/12"
              >
                <span
                  class="sr-only"
                >
                  Info
                </span>
              </th>
              <th
                class="lg:w-1/12 text-right"
              >
                <span
                  class="sr-only"
                >
                  Cost
                </span>
              </th>
              <th
                class="lg:w-1/12 print:hidden"
              >
                <span
                  class="sr-only"
                >
                  Edit
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="align-top"
          >
            <tr>
              <td
                class="align-middle"
              >
                <div
                  class="flex items-start flex-grow"
                >
                  <span
                    class="print:hidden rounded-full h-[18px] w-[18px] md:h-[30px] md:w-[30px] mt-1 md:mt-0 flex-none"
                    style="background-color: rgb(0, 122, 145);"
                  />
                  <span
                    class="ml-[8px]"
                  >
                    Household bills
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right align-middle print:block"
              >
                40
                %
              </td>
              <td
                class="hidden text-center align-middle print:hidden xlg:table-cell"
              >
                <svgr-mock
                  classname=""
                />
              </td>
              <td
                class="text-right flex justify-end items-top align-top font-bold lg:table-cell"
              >
                <span
                  class="pl-4 pr-2 md:px-4 py-2 lg:hidden"
                >
                  £
                  875.00
                </span>
              </td>
              <td
                class="text-right align-top print:hidden lg:hidden"
              >
                <div
                  class="mt-[7px]"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Household bills
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="hidden text-left pl-[24px] justify-start items-top align-top font-bold lg:table-cell"
              >
                <div
                  class="mt-[-14px]"
                >
                  <span
                    class="pl-4 pr-2 py-10"
                  >
                    £
                    875.00
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right justify-end items-top align-top print:hidden lg:table-cell"
              >
                <div
                  class="mt-[-17px] mb-2"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    role="link"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Household bills
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="align-middle"
              >
                <div
                  class="flex items-start flex-grow"
                >
                  <span
                    class="print:hidden rounded-full h-[18px] w-[18px] md:h-[30px] md:w-[30px] mt-1 md:mt-0 flex-none"
                    style="background-color: rgb(245, 103, 39);"
                  />
                  <span
                    class="ml-[8px]"
                  >
                    Living costs
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right align-middle print:block"
              >
                35
                %
              </td>
              <td
                class="hidden text-center align-middle print:hidden xlg:table-cell"
              >
                <svgr-mock
                  classname=""
                />
              </td>
              <td
                class="text-right flex justify-end items-top align-top font-bold lg:table-cell"
              >
                <span
                  class="pl-4 pr-2 md:px-4 py-2 lg:hidden"
                >
                  £
                  771.28
                </span>
              </td>
              <td
                class="text-right align-top print:hidden lg:hidden"
              >
                <div
                  class="mt-[7px]"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Living costs
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="hidden text-left pl-[24px] justify-start items-top align-top font-bold lg:table-cell"
              >
                <div
                  class="mt-[-14px]"
                >
                  <span
                    class="pl-4 pr-2 py-10"
                  >
                    £
                    771.28
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right justify-end items-top align-top print:hidden lg:table-cell"
              >
                <div
                  class="mt-[-17px] mb-2"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    role="link"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Living costs
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="align-middle"
              >
                <div
                  class="flex items-start flex-grow"
                >
                  <span
                    class="print:hidden rounded-full h-[18px] w-[18px] md:h-[30px] md:w-[30px] mt-1 md:mt-0 flex-none"
                    style="background-color: rgb(188, 0, 97);"
                  />
                  <span
                    class="ml-[8px]"
                  >
                    Finance & Insurance
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right align-middle print:block"
              >
                10
                %
              </td>
              <td
                class="hidden text-center align-middle print:hidden xlg:table-cell"
              >
                <svgr-mock
                  classname=""
                />
              </td>
              <td
                class="text-right flex justify-end items-top align-top font-bold lg:table-cell"
              >
                <span
                  class="pl-4 pr-2 md:px-4 py-2 lg:hidden"
                >
                  £
                  100.00
                </span>
              </td>
              <td
                class="text-right align-top print:hidden lg:hidden"
              >
                <div
                  class="mt-[7px]"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Finance & Insurance
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="hidden text-left pl-[24px] justify-start items-top align-top font-bold lg:table-cell"
              >
                <div
                  class="mt-[-14px]"
                >
                  <span
                    class="pl-4 pr-2 py-10"
                  >
                    £
                    100.00
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right justify-end items-top align-top print:hidden lg:table-cell"
              >
                <div
                  class="mt-[-17px] mb-2"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    role="link"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Finance & Insurance
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="align-middle"
              >
                <div
                  class="flex items-start flex-grow"
                >
                  <span
                    class="print:hidden rounded-full h-[18px] w-[18px] md:h-[30px] md:w-[30px] mt-1 md:mt-0 flex-none"
                    style="background-color: rgb(135, 154, 36);"
                  />
                  <span
                    class="ml-[8px]"
                  >
                    Family & Friends
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right align-middle print:block"
              >
                10
                %
              </td>
              <td
                class="hidden text-center align-middle print:hidden xlg:table-cell"
              >
                <svgr-mock
                  classname=""
                />
              </td>
              <td
                class="text-right flex justify-end items-top align-top font-bold lg:table-cell"
              >
                <span
                  class="pl-4 pr-2 md:px-4 py-2 lg:hidden"
                >
                  £
                  100.00
                </span>
              </td>
              <td
                class="text-right align-top print:hidden lg:hidden"
              >
                <div
                  class="mt-[7px]"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Family & Friends
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="hidden text-left pl-[24px] justify-start items-top align-top font-bold lg:table-cell"
              >
                <div
                  class="mt-[-14px]"
                >
                  <span
                    class="pl-4 pr-2 py-10"
                  >
                    £
                    100.00
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right justify-end items-top align-top print:hidden lg:table-cell"
              >
                <div
                  class="mt-[-17px] mb-2"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    role="link"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Family & Friends
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="align-middle"
              >
                <div
                  class="flex items-start flex-grow"
                >
                  <span
                    class="print:hidden rounded-full h-[18px] w-[18px] md:h-[30px] md:w-[30px] mt-1 md:mt-0 flex-none"
                    style="background-color: rgb(0, 13, 61);"
                  />
                  <span
                    class="ml-[8px]"
                  >
                    Travel
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right align-middle print:block"
              >
                2.5
                %
              </td>
              <td
                class="hidden text-center align-middle print:hidden xlg:table-cell"
              >
                <svgr-mock
                  classname=""
                />
              </td>
              <td
                class="text-right flex justify-end items-top align-top font-bold lg:table-cell"
              >
                <span
                  class="pl-4 pr-2 md:px-4 py-2 lg:hidden"
                >
                  £
                  10.00
                </span>
              </td>
              <td
                class="text-right align-top print:hidden lg:hidden"
              >
                <div
                  class="mt-[7px]"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Travel
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="hidden text-left pl-[24px] justify-start items-top align-top font-bold lg:table-cell"
              >
                <div
                  class="mt-[-14px]"
                >
                  <span
                    class="pl-4 pr-2 py-10"
                  >
                    £
                    10.00
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right justify-end items-top align-top print:hidden lg:table-cell"
              >
                <div
                  class="mt-[-17px] mb-2"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    role="link"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Travel
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="align-middle"
              >
                <div
                  class="flex items-start flex-grow"
                >
                  <span
                    class="print:hidden rounded-full h-[18px] w-[18px] md:h-[30px] md:w-[30px] mt-1 md:mt-0 flex-none"
                    style="background-color: rgb(11, 74, 109);"
                  />
                  <span
                    class="ml-[8px]"
                  >
                    Leisure
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right align-middle print:block"
              >
                2.5
                %
              </td>
              <td
                class="hidden text-center align-middle print:hidden xlg:table-cell"
              >
                <svgr-mock
                  classname=""
                />
              </td>
              <td
                class="text-right flex justify-end items-top align-top font-bold lg:table-cell"
              >
                <span
                  class="pl-4 pr-2 md:px-4 py-2 lg:hidden"
                >
                  £
                  10.00
                </span>
              </td>
              <td
                class="text-right align-top print:hidden lg:hidden"
              >
                <div
                  class="mt-[7px]"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Leisure
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="hidden text-left pl-[24px] justify-start items-top align-top font-bold lg:table-cell"
              >
                <div
                  class="mt-[-14px]"
                >
                  <span
                    class="pl-4 pr-2 py-10"
                  >
                    £
                    10.00
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right justify-end items-top align-top print:hidden lg:table-cell"
              >
                <div
                  class="mt-[-17px] mb-2"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    role="link"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Leisure
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`SummarySpendBreakdown component renders correctly as estimate 1`] = `
<div>
  <div
    class="pt-4 lg:pl-7 lg:pt-0 lg:pb-0"
  >
    <h5
      class="text-xl md:text-2xl text-gray-800 font-bold px-4 lg:p-0"
    >
      Spending Breakdown
    </h5>
    <div
      class="flex items-center lg:flex-row flex-col"
    >
      <div
        class="lg:w-4/12 print:hidden w-3/4 px-4 lg:p-0 flex justify-center"
      >
        <div
          class="h-0 pt-[100%] overflow-hidden relative flex-none my-10 w-full"
        >
          <div
            class="after:absolute after:bg-white after:rounded-full after:top-2/4 after:left-2/4 after:-translate-y-2/4 after:-translate-x-2/4 after:h-[55%] after:w-[55%] rounded-full w-full h-full absolute left-0 right-0 top-0 bottom-0"
          />
        </div>
      </div>
      <div
        class="w-full lg:w-8/12"
      >
        <table
          class="w-full lg:text-lg border-separate border-spacing-2 lg:border-spacing-5 lg:ml-10"
        >
          <thead>
            <tr>
              <th
                class="lg:w-9/12"
              >
                <span
                  class="sr-only"
                >
                  Category
                </span>
              </th>
              <th
                class="hidden lg:w-1/12"
              >
                <span
                  class="sr-only"
                >
                  Percentage
                </span>
              </th>
              <th
                class="hidden lg:w-1/12"
              >
                <span
                  class="sr-only"
                >
                  Info
                </span>
              </th>
              <th
                class="lg:w-1/12 text-right"
              >
                <span
                  class="sr-only"
                >
                  Cost
                </span>
              </th>
              <th
                class="lg:w-1/12 print:hidden"
              >
                <span
                  class="sr-only"
                >
                  Edit
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="align-top"
          >
            <tr>
              <td
                class="align-middle"
              >
                <div
                  class="flex items-start flex-grow"
                >
                  <span
                    class="print:hidden rounded-full h-[18px] w-[18px] md:h-[30px] md:w-[30px] mt-1 md:mt-0 flex-none"
                    style="background-color: rgb(0, 13, 61);"
                  />
                  <span
                    class="ml-[8px]"
                  >
                    Travel
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right align-middle print:block"
              >
                2.5
                %
              </td>
              <td
                class="hidden text-center align-middle print:hidden xlg:table-cell"
              >
                <svgr-mock
                  classname=""
                />
              </td>
              <td
                class="text-right flex justify-end items-top align-top font-bold lg:table-cell"
              >
                <span
                  class="text-right bg-gray-100 border-l-4 border-yellow-700 rounded px-2 py-2"
                >
                  £
                  100.00
                </span>
              </td>
              <td
                class="text-right align-top print:hidden lg:hidden"
              >
                <div
                  class="mt-[7px]"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Travel
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="hidden text-left pl-[24px] justify-start items-top align-top font-bold lg:table-cell"
              >
                <div
                  class="mt-[-14px]"
                >
                  <span
                    class="pl-4 pr-2 py-10"
                  >
                    £
                    100.00
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right justify-end items-top align-top print:hidden lg:table-cell"
              >
                <div
                  class="mt-[-17px] mb-2"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    role="link"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Travel
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`SummarySpendBreakdown component renders correctly with tick 1`] = `
<div>
  <div
    class="pt-4 lg:pl-7 lg:pt-0 lg:pb-0"
  >
    <h5
      class="text-xl md:text-2xl text-gray-800 font-bold px-4 lg:p-0"
    >
      Spending Breakdown
    </h5>
    <div
      class="flex items-center lg:flex-row flex-col"
    >
      <div
        class="lg:w-4/12 print:hidden w-3/4 px-4 lg:p-0 flex justify-center"
      >
        <div
          class="h-0 pt-[100%] overflow-hidden relative flex-none my-10 w-full"
        >
          <div
            class="after:absolute after:bg-white after:rounded-full after:top-2/4 after:left-2/4 after:-translate-y-2/4 after:-translate-x-2/4 after:h-[55%] after:w-[55%] rounded-full w-full h-full absolute left-0 right-0 top-0 bottom-0"
          />
        </div>
      </div>
      <div
        class="w-full lg:w-8/12"
      >
        <table
          class="w-full lg:text-lg border-separate border-spacing-2 lg:border-spacing-5 lg:ml-10"
        >
          <thead>
            <tr>
              <th
                class="lg:w-9/12"
              >
                <span
                  class="sr-only"
                >
                  Category
                </span>
              </th>
              <th
                class="hidden lg:w-1/12"
              >
                <span
                  class="sr-only"
                >
                  Percentage
                </span>
              </th>
              <th
                class="hidden lg:w-1/12"
              >
                <span
                  class="sr-only"
                >
                  Info
                </span>
              </th>
              <th
                class="lg:w-1/12 text-right"
              >
                <span
                  class="sr-only"
                >
                  Cost
                </span>
              </th>
              <th
                class="lg:w-1/12 print:hidden"
              >
                <span
                  class="sr-only"
                >
                  Edit
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="align-top"
          >
            <tr>
              <td
                class="align-middle"
              >
                <div
                  class="flex items-start flex-grow"
                >
                  <span
                    class="print:hidden rounded-full h-[18px] w-[18px] md:h-[30px] md:w-[30px] mt-1 md:mt-0 flex-none"
                    style="background-color: rgb(135, 154, 36);"
                  />
                  <span
                    class="ml-[8px]"
                  >
                    Family & Friends
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right align-middle print:block"
              >
                10
                %
              </td>
              <td
                class="hidden text-center align-middle print:hidden xlg:table-cell"
              >
                <svgr-mock
                  classname=""
                />
              </td>
              <td
                class="text-right flex justify-end items-top align-top font-bold lg:table-cell"
              >
                <span
                  class="pl-4 pr-2 md:px-4 py-2 lg:hidden"
                >
                  £
                  100.00
                </span>
                <svgr-mock
                  classname="w-[20px]"
                  fill="green"
                />
              </td>
              <td
                class="text-right align-top print:hidden lg:hidden"
              >
                <div
                  class="mt-[7px]"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Family & Friends
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
            <tr>
              <td
                class="hidden text-left pl-[24px] justify-start items-top align-top font-bold lg:table-cell"
              >
                <div
                  class="mt-[-14px]"
                >
                  <span
                    class="pl-4 pr-2 py-10"
                  >
                    £
                    100.00
                  </span>
                </div>
              </td>
              <td
                class="hidden text-right justify-end items-top align-top print:hidden lg:table-cell"
              >
                <div
                  class="mt-[-17px] mb-2"
                >
                  <button
                    class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none t-link-button tool-nav-submit"
                    formaction="/cost-tab"
                    role="link"
                    type="submit"
                  >
                    <span />
                    Edit
                    <span
                      class="sr-only"
                    >
                      Family & Friends
                    </span>
                    <span />
                  </button>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
